### ISSUE TITLE
imbalanced dataset and accuracy

### ISSUE THREAD

**Issue by ericleonardo** [baturgezici/DTBAPADP#1]
Hi... Analyzing the results in transformer_result2.csv file, I see that training on imbalanced data is affecting accuracy.
Maybe passing class weights to the model will help achieve better accuracy on individual classes.
https://www.tensorflow.org/tutorials/structured_data/imbalanced_data

The confusion matrix from transformer_result2.csv:
![image](https://github.com/baturgezici/price_transformer/assets/26363454/6963a9db-49f3-4624-a121-102e95b200ae)

Congratulations for the great work and thank you for sharing!

**Comment by baturgezici** [comment]
Hello Eric,

Thank you for interest and time in the project, it's great to see people are interested in this topic and project. I used class weights and some other balancing techniques like oversampling and undersampling, but in the end I decided to use oversampling:
https://github.com/baturgezici/price_transformer/blob/12f5e0a36ebc8d5718f1815432d0ab134d60262e/test_transformer.py#L60-L92 and the result of the confusion matrix for ViT architecture can be seen in the paper:
https://ieeexplore.ieee.org/abstract/document/10414094
Again, I will be making updates and upgrades on the project in the near future. In the interim, I would like to investigate and/or discuss any suggestions.

Thank you.

**Comment by ericleonardo** [comment]
Very interesting your work... I would like to experiment more with it.
I will try it with the enhanced [ViTAE transformer](https://github.com/ViTAE-Transformer/ViTAE-Transformer/tree/main/Image-Classification).
For now, I would like to experiment ViT on other data like Forex (EURUSD, etc) also in binary classification.

Please, do you have the Python code for the Data preparation phase (labeling, features, etc)?
I tried the Eclipse IDE with Maven but it is very confusing for non experienced user to edit the CNN-TA Java project.
I got to copy your ViT implementation to Google Colab to facilitate experiments.

Regarding the confusion matrix, it is very interesting that your ViT clearly distinguishes Class 1 from Class 2 perfectly, despite Class 0 dominating both.
Thank you!
